sdks/python/apache_beam/ml/inference/pytorch_test.py

```diff
@@ -0,0 +1,246 @@
+                   dtype="float32")).reshape(-1, 2)
+      expected_predictions = [
+          PredictionResult(ex, pred) for ex,
+          pred in zip(
+              examples,
+              torch.Tensor([f1 * 2.0 + f2 * 3 + 0.5
+                            for f1, f2 in examples]).reshape(-1, 1))
+      ]
+
+      state_dict = OrderedDict([('linear.weight', torch.Tensor([[2.0, 3]])),
+                                ('linear.bias', torch.Tensor([0.5]))])
+      path = os.path.join(self.tmpdir, 'my_state_dict_path')
+      torch.save(state_dict, path)
+
+      model_loader = PytorchModelLoader(
+          state_dict_path=path,
+          model_class=PytorchLinearRegression(input_dim=2, output_dim=1))
+
+      pcoll = pipeline | 'start' >> beam.Create(examples)
+      predictions = pcoll | RunInference(model_loader)
+      assert_that(
+          predictions,
+          equal_to(expected_predictions, equals_fn=_compare_prediction_result))
+
+  def test_pipeline_local_model_with_key(self):
+    with TestPipeline() as pipeline:
+      examples = torch.from_numpy(
+          np.array([1, 5, 3, 10], dtype="float32").reshape(-1, 1))
+      keyed_examples = list(zip(range(len(examples)), examples))
+      expected_values = [
+          PredictionResult(ex, pred) for ex,
+          pred in zip(
+              examples,
+              torch.Tensor([example * 2.0 + 0.5
+                            for example in examples]).reshape(-1, 1))
+      ]
+      expected_predictions = list(zip(range(len(examples)), expected_values))
+
+      state_dict = OrderedDict([('linear.weight', torch.Tensor([[2.0]])),
+                                ('linear.bias', torch.Tensor([0.5]))])
+      path = os.path.join(self.tmpdir, 'my_state_dict_path')
+      torch.save(state_dict, path)
+
+      model_loader = PytorchModelLoader(
+          state_dict_path=path,
+          model_class=PytorchLinearRegression(input_dim=1, output_dim=1))
+
+      pcoll = pipeline | 'start' >> beam.Create(keyed_examples)
+      predictions = pcoll | RunInference(model_loader)
+
+      def _compare_keyed_prediction_result(a, b):
+        key_equal = a[0] == b[0]
+        return (
+            torch.equal(a[1].inference, b[1].inference) and
+            torch.equal(a[1].example, b[1].example) and key_equal)
+
+      assert_that(
+          predictions,
+          equal_to(
+              expected_predictions, equals_fn=_compare_keyed_prediction_result))
+
+  def test_pipeline_gcs_model(self):
+    with TestPipeline() as pipeline:
+      examples = torch.from_numpy(
+          np.array([1, 5, 3, 10], dtype="float32").reshape(-1, 1))
+      expected_predictions = [
+          PredictionResult(ex, pred) for ex,
+          pred in zip(
+              examples,
+              torch.Tensor([example * 2.0 + 0.5
+                            for example in examples]).reshape(-1, 1))
+      ]
+
+      gs_pth = 'gs://apache-beam-ml/pytorch_lin_reg_model_2x+0.5_state_dict.pth'
```

Comment:
I didn't test gcs in my implementation. Does it make sense to break these larger E2E tests into another module instead of the more unit test like tests?

Reply:
You're probably right on separating out the E2E tests. I was thinking that this should be small enough tough to verify the usage of `FileSystems` module though. Perhaps I can break it out when we start adding the E2E testing file.

Reply:
I think it's preferable not to require GCP credentials for a unit test, I'm actually not clear on this, can someone run this unit test and read this GCS path without setting up GCP credentials?